### ISSUE TITLE
Error discovering bulbs

### ISSUE THREAD

**Issue by frakman1** [mclarkk/lifxlan#4]
Although I can get hello_world.py to work on my Mac (Python 2.7 OS X 10.10.3), I can't get it to work on Windows . I am using Python 2.7 on win7-64bit Lenovo Thinkpad machine. I've added 3 seconds to DEFAULT_TIMEOUT but still get an error when attempting to discovering bulbs. Any ideas?
```
C:\Python27\lifxlan-master\lifxlan-master\examples>python hello_world.py 3
Discovering lights...
Traceback (most recent call last):
  File "hello_world.py", line 27, in <module>
    main()
  File "hello_world.py", line 21, in main
    devices = lifx.get_lights()
  File "c:\python27\lib\site-packages\lifxlan\lifxlan.py", line 49, in get_lights
    responses = self.broadcast_with_resp(GetService, StateService)
  File "c:\python27\lib\site-packages\lifxlan\lifxlan.py", line 203, in broadcast_with_resp
    raise WorkflowException("Did not receive {} in response to {}".format(str(response_type), str(msg_type)))
lifxlan.device.WorkflowException: Did not receive <class 'lifxlan.msgtypes.StateService'> in response to <class 'lifxlan.msgtypes.GetService'>

C:\Python27\lifxlan-master\lifxlan-master\examples>
```

**Comment by frakman1** [comment]
I recompiled with verbose set to True for more logs. I hope that helps!

```
C:\Python27\lifxlan-master\lifxlan-master\examples>python hello_world.py

Discovery will go much faster if you provide the number of lights on your LAN:
  python hello_world.py <number of lights on LAN>

Discovering lights...
SEND: GetService
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 100261705
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 1
  Seq Num: 0
  Message Type: 2
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x49', '0xdf', '0xf9', '0x5', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x1', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x2', '0x0', '0x0', '0x0']

RECV: GetService
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 100261705
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 1
  Seq Num: 0
  Message Type: 2
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x49', '0xdf', '0xf9', '0x5', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x1', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x2', '0x0', '0x0', '0x0']

SEND: GetService
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 100261705
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 1
  Seq Num: 0
  Message Type: 2
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x49', '0xdf', '0xf9', '0x5', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x1', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x2', '0x0', '0x0', '0x0']

RECV: GetService
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 100261705
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 1
  Seq Num: 0
  Message Type: 2
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x49', '0xdf', '0xf9', '0x5', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x1', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x2', '0x0', '0x0', '0x0']

SEND: GetService
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 100261705
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 1
  Seq Num: 0
  Message Type: 2
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x49', '0xdf', '0xf9', '0x5', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x1', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x2', '0x0', '0x0', '0x0']

RECV: GetService
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 100261705
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 1
  Seq Num: 0
  Message Type: 2
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x49', '0xdf', '0xf9', '0x5', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x1', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x2', '0x0', '0x0', '0x0']


Found 0 light(s):


C:\Python27\lifxlan-master\lifxlan-master\examples>

```

**Comment by mclarkk** [comment]
Hey there, were you ever able to resolve this issue?

**Comment by frakman1** [comment]
Nope. No progress. Have you made any code changes recently?

**Comment by mclarkk** [comment]
Just a small fix involving labels. I'm trying to find a machine to test the issue on -- I have access to Lenovo ThinkPads, but it's harder for me to find one that boots Windows. I'm wondering if it is an issue with a library that lifxlan relies on. I'll start looking through those to see if there are any known issues.

**Comment by mclarkk** [comment]
Interesting, I just got that error myself today, but I can still control the lights using the broadcast scripts. I'm also having problems getting state from the bulbs using the app, too.

Have you tried using the broadcast on/off/color scripts on the Windows machine that is giving you issues? Do those work, even though you can't discover the devices?

**Comment by frakman1** [comment]
I just tried the broadcast scripts on my problematic windows machine. They did not work. I confirmed that they worked when used on my Mac on the same WiFi. I also confirmed thatI could ping the bulbs from my windows machine. Here is the output:

c:\Python27\lifxlan-master\lifxlan-master\examples>python broadcast_on.py
SEND: LightSetPower
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 2754240116
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 0
  Seq Num: 0
  Message Type: 117
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x74', '0x5e', '0x2a', '0xa4', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x75', '0x0', '0x0', '0x0', '0xff', '0xff', '0x0', '0x0', '0x0', '0x0']

SEND: LightSetPower
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 2754240116
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 0
  Seq Num: 0
  Message Type: 117
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x74', '0x5e', '0x2a', '0xa4', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x75', '0x0', '0x0', '0x0', '0xff', '0xff', '0x0', '0x0', '0x0', '0x0']

SEND: LightSetPower
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 2754240116
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 0
  Seq Num: 0
  Message Type: 117
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x74', '0x5e', '0x2a', '0xa4', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x75', '0x0', '0x0', '0x0', '0xff', '0xff', '0x0', '0x0', '0x0', '0x0']

SEND: LightSetPower
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 2754240116
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 0
  Seq Num: 0
  Message Type: 117
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x74', '0x5e', '0x2a', '0xa4', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x75', '0x0', '0x0', '0x0', '0xff', '0xff', '0x0', '0x0', '0x0', '0x0']

SEND: LightSetPower
  Size: 36
  Origin: 0
  Tagged: 1
  Protocol: 1024
  Source ID: 2754240116
  Target MAC Address: 00:00:00:00:00:00
  Ack Requested: 0
  Response Requested: 0
  Seq Num: 0
  Message Type: 117
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x34', '0x74', '0x5e', '0x2a', '0xa4', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x75', '0x0', '0x0', '0x0', '0xff', '0xff', '0x0', '0x0', '0x0', '0x0']


c:\Python27\lifxlan-master\lifxlan-master\examples>

**Comment by frakman1** [comment]
After making the changes to the routing table as per [this](https://github.com/mclarkk/lifxlan/issues/6) , I got further. Now after dumping out pages of logs, I get this error at the end:

```
SEND: GetVersion
  Size: 36
  Origin: 0
  Tagged: 0
  Protocol: 1024
  Source ID: 2474078350
  Target MAC Address: d0:73:d5:01:73:e9
  Ack Requested: 0
  Response Requested: 1
  Seq Num: 0
  Message Type: 32
  Payload:
    <empty>
  Bytes:
    ['0x24', '0x0', '0x0', '0x14', '0x8e', '0x70', '0x77', '0x93', '0xd0', '0x73', '0xd5', '0x1', '0x73', '0xe9', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x1', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x0', '0x20', '0x0', '0x0', '0x0']

RECV: StateVersion
  Size: 48
  Origin: 1
  Tagged: 0
  Protocol: 1024
  Source ID: 2474078350
  Target MAC Address: d0:73:d5:01:73:e9
  Ack Requested: 0
  Response Requested: 0
  Seq Num: 0
  Message Type: 33
  Payload:
    Vendor: 1
    Reserved: 1
    Version: 6
  Bytes:
    ['0x30', '0x0', '0x0', '0x54', '0x8e', '0x70', '0x77', '0x93', '0xd0', '0x73', '0xd5', '0x1', '0x73', '0xe9', '0x0', '0x0', '0x4c', '0x49', '0x46', '0x58', '0x56', '0x32', '0x0', '0x0', '0x44', '0x19', '0xa4', '0x82', '0xd0', '0x6e', '0x9', '0x14', '0x21', '0x0', '0x0', '0x0', '0x1', '0x0', '0x0', '0x0', '0x1', '0x0', '0x0', '0x0', '0x6', '0x0', '0x0', '0x0']

Traceback (most recent call last):
  File "hello_world.py", line 27, in <module>
    main()
  File "hello_world.py", line 24, in main
    print(d)
  File "c:\python27\lib\site-packages\lifxlan\light.py", line 88, in __str__
    s = self.device_characteristics_str(indent)
  File "c:\python27\lib\site-packages\lifxlan\device.py", line 258, in device_characteristics_str
    s += indent + "Power: {}\n".format(STR_MAP[self.power_level])
KeyError: 65534

c:\Python27\lifxlan-master\lifxlan-master\examples>route print
```

Looks like something is wrong with the printing routines in device.py but I'm not sure how to fix it. Probably a silly matching brace or escape character thing.

**Comment by mclarkk** [comment]
Yeah, I know what that's about. The LIFX protocol docs said that power
level should only be either 0 or 65535, but for some reason your device is
returning a power level of 65534. The code handles that now if you git pull
and reinstall from source.

Glad you're finally getting responses from the lights!

**Comment by frakman1** [comment]
I rolled in that last change and am pleased to say that it now works! Thank you for your stellar support!
I now,finally,  have [this](https://youtu.be/WHCtUvEJXq0) functionality (again) on my desktop computer.

**Comment by mclarkk** [comment]
I love that setup! I'm so glad that some of my code could be a part of something so awesome. Thanks for identifying several bugs and helping to make the library better!

**Comment by frakman1** [comment]
My pleasure. Thank you for writing this library and sharing your hard work with the world.

**Comment by JokerQyou** [comment]
I'm getting this error too, and haven't yet figured out why. I'm using Homebrew Python 2.7.11 on a Macbook, and my bulb is connected to a WiFi network from my only router. I've tried connecting my Macbook via Thunderblot ethernet interface, or directly via the same WiFi. Both did not work for me.

**Comment by mclarkk** [comment]
@JokerQyou I just merged in a bunch of updates that should make things work a lot better. Try pulling and trying again!

**Comment by JokerQyou** [comment]
@mclarkk Maybe it's a little bit late to reply but, I tried your latest code and they worked well. It's not because the library being able to discover my bulb, though. It's because I'm currently using static IP for my bulb, and you added an interface to construct a `Light` instance by manually passing MAC and IP address. I used that and it worked and worked very well, though it still has problem if I switch to `.discover()`.
